File: benchmarks/xlang/src/sieve.R
n <- 10000000
t0 <- proc.time()[["elapsed"]]
composite <- integer(n + 1)
i <- 2
while (i * i <= n) {
  if (composite[i] == 0) {
    j <- i * i
    while (j <= n) {
      composite[j] <- 1L
      j <- j + i
    }
  }
  i <- i + 1
}
count <- 0
for (p in 2:n) if (composite[p] == 0) count <- count + 1
cat("CHECK", count, "\n")
cat("MS", as.integer((proc.time()[["elapsed"]] - t0) * 1000), "\n")

File: benchmarks/xlang/src/sieve.rs
fn main() {
    let n: usize = 10000000;
    let t0 = std::time::Instant::now();
    let mut composite = vec![0u8; n + 1];
    let mut i = 2;
    while i * i <= n {
        if composite[i] == 0 {
            let mut j = i * i;
            while j <= n {
                composite[j] = 1;
                j += i;
            }
        }
        i += 1;
    }
    let count = (2..=n).filter(|&p| composite[p] == 0).count();
    println!("CHECK {}", count);
    println!("MS {}", t0.elapsed().as_millis());
}
